src/Network/Network/AzureFirewallPolicy/RuleGroup/SetAzureFirewallPolicyRuleCollectionGroupCmdlet.cs

```diff
@@ -0,0 +1,191 @@
+        [Parameter(Mandatory = true, ParameterSetName = SetByParentObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public virtual string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The list of rules", ParameterSetName = SetByInputObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSAzureFirewallPolicyRuleCollectionGroupWrapper InputObject { get; set; }
+
+        [Parameter(
+                   Mandatory = true,
+                   ValueFromPipelineByPropertyName = true,
+                   HelpMessage = "Firewall Policy.", ParameterSetName = SetByParentObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSAzureFirewallPolicy FirewallPolicyObject { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.", ParameterSetName = SetByNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public virtual string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource Id of the Rule collection groupy", ParameterSetName = SetByResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public virtual string ResourceId { get; set; }
+
+        [Parameter(
+           Mandatory = true,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "The name of the firewall policy", ParameterSetName = SetByNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public virtual string FirewallPolicyName { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The priority of the rule group")]
+        [ValidateRange(100, 65000)]
+        [Parameter(Mandatory = true, ParameterSetName = SetByNameParameterSet)]
+        [Parameter(Mandatory = true, ParameterSetName = SetByParentObjectParameterSet)]
+        [Parameter(Mandatory = true, ParameterSetName = SetByResourceIdParameterSet)]
+        public uint Priority { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The list of rule collections")]
+        [ValidateNotNullOrEmpty]
+        [Parameter(Mandatory = true, ParameterSetName = SetByNameParameterSet)]
+        [Parameter(Mandatory = true, ParameterSetName = SetByParentObjectParameterSet)]
+        [Parameter(Mandatory = true, ParameterSetName = SetByResourceIdParameterSet)]
+        public PSAzureFirewallPolicyBaseRuleCollection[] RuleCollection { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+            var ruleGroup = new PSAzureFirewallPolicyRuleCollectionGroup();
+            var rcWrapper = new PSAzureFirewallPolicyRuleCollectionGroupWrapper();
+
+            if (this.IsParameterBound(c => c.InputObject))
```

Comment:
Same comments as the `Set-AzFirewallPolicy` cmdlet:
`Priority` and `RuleCollection` should be optional in `-InputObject` parameter set, and if present, they should be used as the value to be set, rather than the value in the `InputObject`

Reply:
I recommand adding a test case for this. For example:
```ps
Get-AzFirewallPolicyRuleCollectionGroup ... | Set-AzFirewallPolicyRuleCollectionGroup -Priority 100
$group = Get-AzFirewallPolicyRuleCollectionGroup ...
Assert-AreEqual 100 $group.Priority
```

Reply:
Fixed. added the test at the end in FirewallPolicyTests.ps1